app/models/pawn.rb

```diff
@@ -4,9 +4,24 @@ def puts_piece_type
     puts "This is a Pawn"
   end
 
-  def valid_move?
-    # put code for how this piece should move here
+  def legal_move?(row,col)
+    current_row = self.row_position
+    current_col = self.col_position
 
+    row_dif = (current_row - row)
+    col_dif = (current_col - col)
+
+
+    #this checks to see if white pawn is moving forward on the board's column
+   if row >= 0 and row <= 7 and col >= 0 and col <= 7
```

Comment:
As I recall, we were pulling this out into a separate piece.rb method.  on_board? or something like that.